python310Packages.manim-voiceover 0.3.0

FaustXVI opened the pull request:
###### Description of changes

Linked to my package request #217496 and starting from @SuperSandro2000 version of manim (#212484)

###### Things done

Work in progress, got gtts working now trying to get recording

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

mweinelt commented:
Please read CONTRIBUTING.md on how the commit messages should be written. Also, please add an explanation why we move from torch to torch-bin.

FaustXVI commented:
> Please read CONTRIBUTING.md on how the commit messages should be written. Also, please add an explanation why we move from torch to torch-bin.

Hi, yes my commit messages are all wrong, sorry.

I am trying to make it work for the moment and will squash and make a proper commit message once done. That's the reason I marked the PR as Draft. My idea was to show I started working on it so if someone wants to help they don't start from scratch.

It's definitely not ready for review for now.

Sorry if that makes a mess. I can close this PR and reopen one clean once done if you prefer.

For `torch` vs `torch-bin`, my goal is to not change but I am having conflicts for now. Hopefully I'll revert that change.

Thanks for your reactivity :)

FaustXVI committed:
maintainers: add FaustXVI

junjihashimoto commented:
I think that torch-bin is used to use torchaudio-bin. No torchaudio yet.

FaustXVI commented:
@junjihashimoto, yes indeed.
I fixed it with an override.
A first version of this PR is almost ready, I still have to add myself as a maintainer, which I'll do as soon as my other PR is merged.
If you're interested in testing it though, please feel free.

FaustXVI committed:
pythonPackages.humanhash3 : init at version 0.0.6

FaustXVI committed:
pythonPackages.sox : init at version 1.4.1

FaustXVI committed:
pythonPackages.stable-ts: init at version 1.1.1b0

FaustXVI committed:
pythonPackages.manim-voiceover: init at version 0.3.0

User still have to add following dependencies to their environment :
sox gst_all_1.gstreamer intltool gobject-introspection gst_all_1.gst-plugins-base gst_all_1.gst-plugins-good

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -4769,6 +4769,11 @@
     name = "Jakub Okoński";
   };
+  FaustXVI = {
+    name = "Xavier Detant";
+    github = "FaustXVI";
+    githubId = 1016863;
```
Ideally we should have any other method than github to reach out to you.

FaustXVI replied:
Is my github anonymous email ok ?

SuperSandro2000 commented on pkgs/development/python-modules/humanhash3/default.nix:
```diff
@@ -0,0 +1,26 @@
+    extension = "tar.gz";
+    hash = "sha256-ia4V9gNL2wflOsKy4kbMcTGUB2bSsrcLc5jdJ8ZAsas=";
+  };
+
```
```suggestion

  # has no tests
  doCheck = false;

```

SuperSandro2000 commented on pkgs/development/python-modules/manim-voiceover/default.nix:
```diff
@@ -0,0 +1,83 @@
+  version = "0.3.0";
+  disabled = pythonOlder "3.7";
+
+  extras = [ "gtts" "recorder" ];
```
This should probably be in passthru.optional-dependencies

FaustXVI replied:
Done

SuperSandro2000 commented on pkgs/development/python-modules/manim-voiceover/default.nix:
```diff
@@ -0,0 +1,83 @@
+    humanhash3
+    sox
+    python-dotenv
+    pip
```
We really need pip?

FaustXVI replied:
I removed it and it still build, so I assume no.
I just took what was in the dependency file of the project.

FaustXVI replied:
I put it back because it build but doesn't run properly.

SuperSandro2000 commented on pkgs/development/python-modules/manim-voiceover/default.nix:
```diff
@@ -0,0 +1,83 @@
+  nativeCheckInputs = [
+    pytest-xdist
+    pytestCheckHook
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/manim-voiceover/default.nix:
```diff
@@ -0,0 +1,83 @@
+  ];
+
+  doCheck = false;
+
+
+  meta = with lib; {
```
```suggestion

  meta = with lib; {
```

SuperSandro2000 commented on pkgs/development/python-modules/manim-voiceover/default.nix:
```diff
@@ -0,0 +1,83 @@
+
+  ];
+
+  doCheck = false;
```
Not sure what the test does https://github.com/ManimCommunity/manim-voiceover/blob/main/tests/test_install.py

FaustXVI replied:
Tests just test the import like `pythonImportsCheck` so I am ignoring them.

SuperSandro2000 commented on pkgs/development/python-modules/stable-ts/default.nix:
```diff
@@ -0,0 +1,47 @@
+    numpy
+  ];
+
+  doCheck = false;
```
```suggestion
  # has no tests
  doCheck = false;
```

FaustXVI commented on pkgs/development/python-modules/svgelements/default.nix:
```diff
@@ -0,0 +1,38 @@
+    scipy
+  ];
+
+  pythonImportsCheck = [ "svgelements" ];
```
Done, thanks

FaustXVI commented on pkgs/development/python-modules/svgelements/default.nix:
```diff
@@ -0,0 +1,38 @@
+    sha256 = "sha256-Ic4y8YjXlo2gf1hdiBQcDNhjgEmmiBPWnH65QfoJ+tY=";
+  };
+
+  propagatedBuildInputs = [ ];
+
```
I'll rebase on master as soon as your PR is merged.
I shouldn't have any change in this file.

FaustXVI commented on pkgs/development/python-modules/sox/default.nix:
```diff
@@ -0,0 +1,35 @@
+    typing-extensions
+  ];
+
+  doCheck = false;
```
I am working on this but I got a lot of tests looking for files in `/build/sox-1.4.1/tests/data` but are not found.
Test code is using `os.path.dirname(__file__)` which I assume returns `/build/sox-1.4.1/tests` which seems to not be the correct value.

Since I am not a python dev *and* new to nix packaging, I have no idea on how to fix that.

Any help is welcome.

gaelreyrol committed:
mqttx: init at 1.9.8

OPNA2608 committed:
lomiri.lomiri-sounds: init at 22.02

fabaff committed:
python311Packages.impacket: enable tests

marsam committed:
rufo: 0.16.2 -> 0.17.0

Diff: https://github.com/ruby-formatter/rufo/compare/v0.16.2...v0.17.0

Changelog: https://github.com/ruby-formatter/rufo/blob/v0.17.0/CHANGELOG.md

GaetanLepage committed:
vimPlugins.cmp-tabby: init at 2023-11-21

r-ryantm committed:
yutto: 2.0.0b32 -> 2.0.0b33

RatCornu committed:
maintainers: add ratcornu

RatCornu committed:
suwayomi-server: init at 0.7.0

RatCornu committed:
nixos/suwayomi-server: init at 0.7.0

RatCornu committed:
nixos/suwayomi-server: add nixos tests

RatCornu committed:
nixos/suwayomi-server: add release note

r-ryantm committed:
sentry-cli: 2.23.2 -> 2.25.2

fabaff committed:
python311Packages.asyncclick: 8.1.3.2 -> 8.1.7.1

Diff: https://github.com/python-trio/asyncclick/compare/refs/tags/8.1.3.2...8.1.7.1

Changelog: https://github.com/python-trio/asyncclick/blob/8.1.7.1/CHANGES.rst

fabaff committed:
python311Packages.lmcloud: init at 0.4.35

Library to interface with La Marzocco's cloud

https://github.com/zweckj/lmcloud

fabaff committed:
python311Packages.sqlglot: refactor

fabaff committed:
python311Packages.sqlglot: 17.14.2 -> 20.9.0

Diff: https://github.com/tobymao/sqlglot/compare/refs/tags/v17.14.2...v20.9.0

Changelog: https://github.com/tobymao/sqlglot/blob/v20.9.0/CHANGELOG.md

GaetanLepage committed:
vimPlugins: update on 2024-01-19

GaetanLepage committed:
vimPlugins.nvim-treesitter: update grammars

r-ryantm committed:
distgen: 1.5 -> 1.17

r-ryantm committed:
python311Packages.django-import-export: 3.3.5 -> 3.3.6

FaustXVI committed:
Add hacky way to get dotenv to a version expected by manim voiceover

FaustXVI commented on pkgs/development/python-modules/python-dotenv-0_21/default.nix:
```diff
@@ -0,0 +1,43 @@
+
+buildPythonPackage rec {
+  pname = "python-dotenv";
+  version = "0.21.0";
```
I really dislike doing this but manim-voiceover is expecting a version < 0.22 and >= 0.21 and I don't know how to do that properly.
This file is taken from commit 000bf335745ae7af84e17ff6be401867e9180a5d of nixpkgs
